src/lib/runner.js

```diff
@@ -4,6 +4,15 @@ import path from 'path';
 import { spawn } from 'child_process';
 
 class Runner {
+  // JS strings can be up to 2^53 - 1 in length,
+  // so no need to worry about overflow
+  // https://262.ecma-international.org/16.0/index.html#sec-ecmascript-language-types-string-type
+  #stdOutBuffer = '';
+  #stdErrBuffer = '';
+
+  #stdOutSubscribers = [];
```

Comment:
how come an array?

Reply:
No, I admit it is not. My intention was to better support future use cases using a flexible implementation.